q2_types/_feature_data.py

```diff
@@ -0,0 +1,183 @@
+            while count < 10:
+                line = fh.readline()
+                if line == '':
+                    break
+                elif line.startswith('#'):
+                    continue
+                else:
+                    cells = line.split('\t')
+                    if len(cells) < 2:
+                        return False
+                    count += 1
+            return False if count == 0 else True
+
+
+class FASTAFormat(FileFormat):
+    name = 'fasta'
+
+    @classmethod
+    def sniff(cls, filepath):
+        sniffer = skbio.io.io_registry.get_sniffer('fasta')
+        return sniffer(filepath)[0]
+
+
+class AlignedDNAFASTAFormat(FileFormat):
+    name = 'aligned-dna-fasta'
+
+    @classmethod
+    def sniff(cls, filepath):
+        sniffer = skbio.io.io_registry.get_sniffer('fasta')
+        if sniffer(filepath)[0]:
+            generator = skbio.io.read(filepath, constructor=skbio.DNA,
+                                      format='fasta')
+            try:
+                initial_length = len(next(generator))
+                for seq, _ in zip(generator, range(4)):
+                    if len(seq) != initial_length:
+                        return False
+                return True
+            # ValueError raised by skbio if there are invalid DNA chars.
+            except (StopIteration, ValueError):
+                return False
+        else:
+            return False
+
+
+taxonomy_data_layout = DataLayout('taxonomy', 1)
+taxonomy_data_layout.register_file('taxonomy.tsv', TaxonomyFormat)
+
+sequences_data_layout = DataLayout('sequences', 1)
+sequences_data_layout.register_file('sequences.fasta', FASTAFormat)
+
+aligned_dna_sequences_data_layout = DataLayout('aligned-dna-sequences', 1)
+aligned_dna_sequences_data_layout.register_file('aligned-dna-sequences.fasta',
+                                                AlignedDNAFASTAFormat)
+
+
+def taxonomy_to_pandas_series(data_dir):
+    filepath = os.path.join(data_dir, 'taxonomy.tsv')
+    # Using read_csv for access to comment parameter, and set index_col,
+    # header, and parse_dates to Series.from_csv default settings for
+    # better handling of round-trips
+    df = pd.read_csv(filepath, sep='\t', comment='#', index_col=0, header=0,
+                     parse_dates=True, skip_blank_lines=True)
+    return df.iloc[:, 0]
+
+
+def taxonomy_to_qiime_metadata(data_dir):
+    filepath = os.path.join(data_dir, 'taxonomy.tsv')
+    df = pd.read_csv(filepath, sep='\t', comment='#', index_col=0, header=0,
+                     parse_dates=True, skip_blank_lines=True)
+    return qiime.Metadata(df)
+
+
+def pandas_series_to_taxonomy(view, data_dir):
+    filepath = os.path.join(data_dir, 'taxonomy.tsv')
+    view.to_csv(filepath, sep='\t', header=True, index=True)
+
+
+def qiime_metadata_to_taxonomy(view, data_dir):
+    filepath = os.path.join(data_dir, 'taxonomy.tsv')
+    df = view.to_dataframe()
+    df.to_csv(filepath, sep='\t', header=True, index=True)
```

Comment:
When we have a taxonomy assigner (e.g. analogous to QIIME 1's assign_taxonomy.py), it will need to return a view type that supports an arbitrary number of columns (so returning a `pd.Series` isn't sufficient). I was imagining the taxonomy assigner would return a `qiime.Metadata`. Does this approach make sense to you or is there a better way to handle this?